src/ZipkinBundle/Components/Messenger/ZipkinSendHandler.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+
+namespace ZipkinBundle\Components\Messenger;
+
+use Symfony\Component\Messenger\Envelope;
+
+use Zipkin\Kind;
+use Zipkin\Propagation\B3;
+use Zipkin\Tags;
+use Zipkin\Tracing;
+
+class ZipkinSendHandler implements ZipkinHandlerInterface
+{
+    /**
+     * @var \Zipkin\Tracer
+     */
+    private $tracer;
+
+    public function __construct(Tracing $tracing)
+    {
+        $this->tracer = $tracing->getTracer();
+    }
+
+    public function handle(Envelope $envelope): Envelope
+    {
+        if ($envelope->all(ZipkinStamp::class)) {
```

Comment:
As far as I understand if we find a stamp we keep going, otherwise we create a span? I kind of not understand that, maybe a comment ellaborating it would be helpful?

Reply:
I'm checking that anyone has already added the stamp. If the stamp is missing so we create a new Span and fill the span.